File: 백준/Silver/10828. 스택/스택.cc
#include <iostream>
#include <string>
#include <stack>
using namespace std;

int main() {
	int n;
	string command;
	int num;
	int result;	//출력하는 결과
	stack <int> s;
	cin >> n;

	for (int i = 0; i < n; i++) {
		cin >> command;
		if (command == "push") {
			cin >> num;
			s.push(num);
		}

		else if (command == "pop") {
			if (s.empty()) {
				cout << -1 << endl;
			}
			else {
				result = s.top();
				cout << result << endl;
				s.pop();
			}
		}

		else if (command == "size") {
			cout << s.size() << endl;
		}

		else if (command == "empty") {
			if (s.empty()) {
				cout << 1 << endl;
			}
			else {
				cout << 0 << endl;
			}
		}

		else if (command == "top") {
			if (s.empty()) {
				cout << -1 << endl;
			}
			else {
				result = s.top();
				cout << result << endl;
			}
		}
	}
	return 0;
}

File: 백준/Silver/10828. 스택/스택.java
import java.util.*;
import java.io.IOException;

class Main {
    public static void main(String[] args) throws IOException {
        Stack<Integer> stack = new Stack<>();
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();

        for (int i = 0; i < n; i++) {
            String str = sc.next();
            
            switch(str) {
                case "push":
                    int num = sc.nextInt();
                    stack.push(num); break;
                case "pop":
                    if (stack.isEmpty()) System.out.println(-1);
                    else System.out.println(stack.pop());
                    break;
                case "size":
                    System.out.println(stack.size()); break;
                case "empty":
                    if (stack.isEmpty()) System.out.println(1);
                    else System.out.println(0);
                    break;
                case "top":
                    if (stack.isEmpty()) System.out.println(-1);
                    else System.out.println(stack.peek());
                    break;
            }
        }
    }
}